app/jobs/pollable_job_wrapper.rb

```diff
@@ -59,6 +64,18 @@ def persist_warnings(job)
       def save_error(api_error, job)
         find_pollable_job(job).each do |pollable_job|
           pollable_job.update(cf_api_error: api_error)
+        rescue Sequel::DatabaseError => ex
+          begin
+            m = /value too long for type character varying\((\d+)\)/.match(ex.message)
+            limit = m ? m[1].to_i - 1 : 15_999
```

Comment:
what are the cases where the character size limit is not 16,000?

Reply:
don't know.  With the change in `convert_to_v3_api_error` I no longer was hitting any of the exceptions in `save_error`, but left those exception handlers in so they could be logged. If you think they should be dropped (and the yaml truncator files removed) I'm ok with that

Reply:
Yeah, if this code isn't getting hit anymore, I think it makes sense to remove it.  Seems like this code would only get hit if someone manually altered the DB.

One possible alternative that would combine the two approaches--attempt to save the job, catch the error, and only then cut the backtrace in half.  That way we wouldn't need to know the magic `16000` number

Reply:
I might have something, that still calls for the yaml truncator....

Reply:
Yup, did that more or less. The refactor puts the exception backtrace cutting and yaml-checking in one place, although only the postgres error message gives the max field width. Note that the travis PR is now failing because of an error in master (discussed on the vat channel), but the push itself is fine

Reply:
I dropped the whole yaml truncator because it isn't needed for this case, and probably will never be. No one's going to emit an exception with a 12K message.